code/modules/mob/mob_helpers.dm

```diff
@@ -346,6 +346,29 @@ proc/slur(phrase)
 		newphrase+="[newletter]";counter-=1
 	return newphrase
 
+proc/roboslur(phrase) //for robutts
+	phrase = html_decode(phrase)
+	var/leng=lentext(phrase)
+	var/counter=lentext(phrase)
+	var/newphrase=""
+	var/newletter=""
+	while(counter>=1)
+		newletter=copytext(phrase,(leng-counter)+1,(leng-counter)+2)

```

Comment:
```counter``` gets smaller by 1 every iteration.
so while it starts at 0+1, 0+2
it slowly becomes:
1+1, 1+2
2+1, 2+2
3+1, 3+2
as ```leng``` is never changed.